File: app/MultiMarket/go/data/flea/irank/irank.go
// Copyright 13-Sep-2020 ºDeme
// GNU General Public License - V3 <http://www.gnu.org/licenses/>

// Investors ranking entry.
package irank

import (
	"github.com/dedeme/MultiMarket/data/flea/investor"
	"github.com/dedeme/golib/json"
)

type T struct {
	date    string
	ranking []*investor.T
}

// Cretes a new 'irank'.
//    date   : Ranking date
//    ranking: Investors list.
func New(date string, ranking []*investor.T) *T {
	return &T{date, ranking}
}

func (r *T) Date() string {
	return r.date
}

func (r *T) Ranking() []*investor.T {
	return r.ranking
}

func (r *T) ToJs() json.T {
	var rk []json.T
	for _, e := range r.ranking {
		rk = append(rk, e.ToJs())
	}
	return json.Wa([]json.T{
		json.Ws(r.date),
		json.Wa(rk),
	})
}

func (r *T) ToJsClient() json.T {
	var rk []json.T
	for _, e := range r.ranking {
		rk = append(rk, e.ToJsClient())
	}
	return json.Wa([]json.T{
		json.Ws(r.date),
		json.Wa(rk),
	})
}

func FromJs(js json.T) *T {
	a := js.Ra()
	return &T{
		a[0].Rs(),
		investor.FromJsList(a[1].Ra()),
	}
}

File: app/MMarket_old202009/js/src/data/flea/Irank.js
// Copyright 31-May-2020 ºDeme
// GNU General Public License - V3 <http://www.gnu.org/licenses/>

import Investor from "./Investor.js";

/**
    Ranking of Investors.
**/
export default class Irank {
  /* .
  _rc_ Irank: from
    date: string
    invs: !Array<!Investor>
  */

  /*--*/
  /**
      @param {string} date
      @param {!Array<!Investor>} invs
  **/
  constructor (date, invs) {

    /**
        @private
        @type {string}
    **/
    this._date = date;

    /**
        @private
        @type {!Array<!Investor>}
    **/
    this._invs = invs;

  }

  /**
      @return {string}
  **/
  get date () {
    return this._date;
  }

  /**
      @return {!Array<!Investor>}
  **/
  get invs () {
    return this._invs;
  }

  /**
      @param {!Array<?>} serial
      @return {!Irank}
  **/
  static fromJs (serial) {
    return new Irank(
      serial[0],
      serial[1].map(e =>
        Investor.fromJs(e)
      )
    );
  }
  /*--*/
}
